### PR TITLE
Linking POIs to POI information activity, improving the UI and linking buttons

### PR DESCRIPTION
**Pull request by MarcelTorne**
In this PR I am improving the UI of the POI information activity by adding a link preview, a button that takes you to the directions in google maps and more...

I am also linking the Chat, Review, Forum Activities to this activity.

Finally, from now on, when clicking on a POI you will be redirected to this information activity.

Captura de Pantalla 2021-05-20 a les 14 06 53

### CODE REVIEW

**Comment by tberkane** [comment]
For me, the phone button is not working, I think there is a missing permission.

**Comment by aman3014** [comment]
As for the back buttons in the chat/review/forum, now I'm realising that the back button might be important since the normal back button just takes u to the previous activity.

So with a flow like Info -> Chat -> Reviews -> Forum you would need to press the back button 4 times to get back to the map. 

For now, I suggest adding the back button to the map activity as the parent for all four poi activities since the map doesn't need an intent.

**Comment by MarcelTorne** [comment]
> As for the back buttons in the chat/review/forum, now I'm realising that the back button might be important since the normal back button just takes u to the previous activity.
> 
> So with a flow like Info -> Chat -> Reviews -> Forum you would need to press the back button 4 times to get back to the map.
> 
> For now, I suggest adding the back button to the map activity as the parent for all four poi activities since the map doesn't need an intent.

Sounds good! Added!

File: app/src/main/java/com/github/epfl/meili/poi/PoiInfoFragment.kt [hunk 1]
```diff
@@ -1,30 +1,47 @@
...
 import com.google.android.libraries.places.api.net.*
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import io.github.ponnamkarthik.richlinkpreview.RichLinkViewSkype
+import io.github.ponnamkarthik.richlinkpreview.ViewListener
 
 /**
  * Fragment to be displayed inside of PoiActivity and which contains basic info about POI
  */
-class PoiInfoFragment(val poi: PointOfInterest) : Fragment() {
+class PoiInfoFragment(val poi: PointOfInterest) : Fragment() { //TODO verify that everything working properly when better connection
```
Comment (aman3014) [inline app/src/main/java/com/github/epfl/meili/poi/PoiInfoFragment.kt:32]: Is this a leftover TODO or something we should do in the future?

**Review by aman3014 — APPROVED** [review]
Looks great, thanks for your work !

Just left a couple of minor comments that you can fix before merging.

File: app/src/main/java/com/github/epfl/meili/poi/PoiInfoFragment.kt (hunk 1 above)
Comment (MarcelTorne) [reply app/src/main/java/com/github/epfl/meili/poi/PoiInfoFragment.kt:32]: Leftover thanks!

File: app/src/main/java/com/github/epfl/meili/map/MapActivity.kt [hunk 2]
```diff
@@ -19,6 +19,7 @@ import com.github.epfl.meili.database.FirestoreDatabase
 import com.github.epfl.meili.home.Auth
 import com.github.epfl.meili.models.PointOfInterest
 import com.github.epfl.meili.photo.CameraActivity
+import com.github.epfl.meili.poi.PoiActivity
 import com.github.epfl.meili.poi.PoiServiceCached
 import com.github.epfl.meili.posts.forum.ForumActivity
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/map/MapActivity.kt:24]: Unused import

File: app/src/main/java/com/github/epfl/meili/messages/ChatLogActivity.kt [hunk 3]
```diff
@@ -14,6 +14,7 @@ import com.github.epfl.meili.map.MapActivity
 import com.github.epfl.meili.models.ChatMessage
 import com.github.epfl.meili.models.PointOfInterest
 import com.github.epfl.meili.models.User
+import com.github.epfl.meili.poi.PoiActivity
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/messages/ChatLogActivity.kt:17]: Unused import

File: app/src/main/java/com/github/epfl/meili/poi/PoiActivity.kt [hunk 4]
```diff
@@ -8,12 +8,11 @@ import androidx.viewpager2.widget.ViewPager2
 import com.github.epfl.meili.R
 import com.github.epfl.meili.map.MapActivity
 import com.github.epfl.meili.models.PointOfInterest
+import com.github.epfl.meili.util.MenuActivity
 
-class PoiActivity : AppCompatActivity() {
+class PoiActivity : MenuActivity(R.menu.nav_poi_info_menu) {
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/poi/PoiActivity.kt:13]: Now that we are not using the viewpager anymore, we can combine this and PoiInfoFragment into a single activity I think

File: app/src/main/java/com/github/epfl/meili/poi/PoiInfoFragment.kt [hunk 5]
```diff
@@ -34,64 +51,124 @@ class PoiInfoFragment(val poi: PointOfInterest) : Fragment() {
...
+                val openStatusInfo = view.findViewById<TextView>(R.id.openStatusTextView)
+                openStatusInfo.text = openText
+
+                val infoTextView = view.findViewById<TextView>(R.id.infoTextView)
+                "${place.address}\n".also {
+                    infoTextView.text = it
+                }
+
+                if (place.websiteUri != null && place.websiteUri.toString().isNotEmpty()) {
+                    val richLinkView = view.findViewById<RichLinkViewSkype>(R.id.richLinkView)
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/poi/PoiInfoFragment.kt:104]: This is not working very well on my phone, on some pois it just shows as a white rectangle

File: app/src/main/java/com/github/epfl/meili/poi/PoiInfoFragment.kt [hunk 6]
```diff
@@ -34,64 +51,124 @@ class PoiInfoFragment(val poi: PointOfInterest) : Fragment() {
...
+                }
 
-            val poiImageView = view.findViewById<ImageView>(R.id.poiImageView)
+                takeMeThereButton.visibility = VISIBLE
 
-            val metada = place.photoMetadatas
-            if (!(metada == null || metada.isEmpty())) {
+                takeMeThereButton.setOnClickListener {
+                    val gmmIntentUri =
+                            Uri.parse("google.navigation:q=${place.address}")
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/poi/PoiInfoFragment.kt:121]: You could use string constants here

File: app/src/main/java/com/github/epfl/meili/poi/PoiInfoFragment.kt [hunk 7]
```diff
@@ -34,64 +51,124 @@ class PoiInfoFragment(val poi: PointOfInterest) : Fragment() {
...
+                }
+            }
+
+    private fun launchCallIntent(phoneNumber: String?) {
+        if (phoneNumber != null) {
+            if (ContextCompat.checkSelfPermission(MainApplication.applicationContext(), Manifest.permission.CALL_PHONE) != PackageManager.PERMISSION_GRANTED) {
+                ActivityCompat.requestPermissions(requireActivity(), arrayOf(Manifest.permission.CALL_PHONE), REQUEST_CODE)
+            } else {
+                val callIntent = Intent(Intent.ACTION_CALL)
+                callIntent.data = Uri.parse("tel:${phoneNumber}")
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/poi/PoiInfoFragment.kt:154]: Also here and below

File: app/src/main/java/com/github/epfl/meili/posts/forum/ForumActivity.kt [hunk 8]
```diff
@@ -17,10 +17,12 @@ import com.github.epfl.meili.database.AtomicPostFirestoreDatabase
 import com.github.epfl.meili.database.FirestoreDatabase
 import com.github.epfl.meili.home.Auth
 import com.github.epfl.meili.map.MapActivity
+import com.github.epfl.meili.map.MapActivity.Companion.POI_KEY
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/posts/forum/ForumActivity.kt:20]: Unused import

File: app/src/main/java/com/github/epfl/meili/review/ReviewsActivity.kt [hunk 9]
```diff
@@ -1,5 +1,6 @@
 package com.github.epfl.meili.review
 
+import android.content.Intent
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/review/ReviewsActivity.kt:3]: Some unused imports

File: app/src/main/java/com/github/epfl/meili/util/MenuActivity.kt [hunk 10]
```diff
@@ -42,13 +45,15 @@ open class MenuActivity(private val menuRes: Int): AppCompatActivity() {
             R.id.menu_reviews -> ReviewsActivity::class.java
             R.id.menu_chat -> ChatLogActivity::class.java
             R.id.menu_forum -> ForumActivity::class.java
-            else -> MapActivity::class.java
+            else -> PoiActivity::class.java
         }
+        val poi = intent.getParcelableExtra<PointOfInterest>(
+                MapActivity.POI_KEY
+        )
+        Log.d("Menu Activity", poi.toString())
```
Comment (tberkane) [inline app/src/main/java/com/github/epfl/meili/util/MenuActivity.kt:53]: Is this leftover from debugging?

File: app/src/main/res/layout/fragment_poi_info.xml [hunk 11]
```diff
@@ -7,14 +7,41 @@
     tools:context=".poi.PoiInfoFragment">
```
Comment (tberkane) [inline app/src/main/res/layout/fragment_poi_info.xml:8]: You have hardcoded strings and missing content descriptions in the resource files

**Review by tberkane — CHANGES_REQUESTED** [review]
Looks way better than before, thanks!

File: app/src/main/java/com/github/epfl/meili/poi/PoiInfoFragment.kt (hunk 5 above)
Comment (MarcelTorne) [reply app/src/main/java/com/github/epfl/meili/poi/PoiInfoFragment.kt:104]: Yeah cause some websites don't allow the preview option. But you can always click and you'll get redirected to the website. If we have better ideas I propose we add them for next weeks task. What do you think?

**Comment by MarcelTorne** [comment]
@tberkane Thank you for all your comments, they were really useful!

**Review by tberkane — APPROVED** [review]

**Pull request merged**